src/main/java/com/epam/ta/reportportal/core/bts/handler/impl/UpdateBugTrackingSystemHandlerImpl.java

```diff
@@ -145,22 +135,67 @@ public OperationCompletionRS updateBugTrackingSystem(UpdateBugTrackingSystemRQ u
 	}
 
 	@Override
-	public OperationCompletionRS integrationConnect(UpdateBugTrackingSystemRQ updateRequest, Long integrationId,
+	public OperationCompletionRS testIntegrationConnection(BtsConnectionTestRQ connectionTestRQ, Long integrationId,
 			ReportPortalUser.ProjectDetails projectDetails) {
-		Integration integration = integrationRepository.findById(integrationId)
-				.orElseThrow(() -> new ReportPortalException(INTEGRATION_NOT_FOUND, integrationId));
 
-		Integration details = new BugTrackingSystemBuilder().addUrl(updateRequest.getUrl())
-				.addBugTrackingProject(updateRequest.getProject())
-				.get();
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(ErrorType.PROJECT_NOT_FOUND, projectDetails.getProjectName()));
+
+		Integration integration = getBugTrackingSystemHandler.getEnabledByProjectIdAndId(projectDetails, integrationId).orElseGet(() -> {
```

Comment:
i see the duplicates of getting integration logic in several places. CreateTicketHandler and down here too.
maybe we can move it somewhere?
i think, that we can move it in getBugTrackingSystemHandler
create method getIntegartion(bla bla) and move this logic there

so, when we are reading code what we would see. getIntegration (okay, this method gets an integration)
if we need to understand logic we go inside and see that (getProjectIntegration or else getGlobal)
so, as i think that it is pretty simple to understand

Reply:
done